### PR TITLE
support preempt with priority plugin alone

### PR DESCRIPTION
**Pull request by Thor-wl**
fix: https://github.com/volcano-sh/volcano/issues/1720 
Signed-off-by: Thor-wl <13164644535@163.com>

### CODE REVIEW

File: pkg/scheduler/plugins/priority/priority.go [hunk 1]
```diff
@@ -113,6 +113,26 @@ func (pp *priorityPlugin) OnSessionOpen(ssn *framework.Session) {
...
+			}
+		}
+
+		return nil
+	}
+	ssn.AddJobValidFn(pp.Name(), validJobFn)
+
+	jobStarvingFn := func(obj interface{}) bool {
+		ji := obj.(*api.JobInfo)
+		return ji.ReadyTaskNum() < int32(len(ji.Tasks))
```
Comment (william-wang) [inline pkg/scheduler/plugins/priority/priority.go:132]: Please take a look whether we need to consider the WaitingTaskNum here as well?

File: pkg/scheduler/plugins/priority/priority.go [hunk 2]
```diff
@@ -113,6 +113,26 @@ func (pp *priorityPlugin) OnSessionOpen(ssn *framework.Session) {
 	}
 
 	ssn.AddPreemptableFn(pp.Name(), preemptableFn)
+
+	validJobFn := func(obj interface{}) *api.ValidateResult {
```
Comment (william-wang) [inline pkg/scheduler/plugins/priority/priority.go:117]: If there is not special things to do in validJobFn. It's better to remove this function.

File: pkg/scheduler/plugins/priority/priority.go (hunk 1 above)
Comment (huone1) [reply pkg/scheduler/plugins/priority/priority.go:132]: it is the same as the code in gang plugin。 it is repeat。 I think it is better to modify the default value of the function JobStarving and change from false to true

File: pkg/scheduler/plugins/priority/priority.go (hunk 2 above)
Comment (huone1) [reply pkg/scheduler/plugins/priority/priority.go:117]: > If there is not special things to do in validJobFn. It's better to remove this function.

I also agree to delete it.

File: pkg/scheduler/plugins/priority/priority.go (hunk 1 above)
Comment (Thor-wl) [reply pkg/scheduler/plugins/priority/priority.go:132]: > Please take a look whether we need to consider the WaitingTaskNum here as well?

Yes, It is needed. Let me add it.

Comment (Thor-wl) [reply pkg/scheduler/plugins/priority/priority.go:132]: > it is the same as the code in gang plugin。 it is repeat。 I think it is better to modify the default value of the function JobStarving and change from false to true

No, they are not same. `minAvailable` is not considered here.

**Review by william-wang — APPROVED** [review]
/lgtm
/approve

**Pull request merged**